wrong check if environment variable HIP_PATH is defined

in the root CMakeLists.txt, currently in the develop branch on lines 58--62, the HIP_PATH cmake variable is (supposed to be) set depending on the HIP_PATH environment variable:
```
if( NOT DEFINED $ENV{HIP_PATH})
    set( HIP_PATH "/opt/rocm/hip" )
else( )
    set (HIP_PATH $ENV{HIP_PATH} )
endif( )
```
but this is the wrong way of checking if an environment variable is defined. See
[https://cmake.org/cmake/help/latest/variable/ENV.html](https://cmake.org/cmake/help/latest/variable/ENV.html)
or
[https://cmake.org/pipermail/cmake/2011-October/046706.html](https://cmake.org/pipermail/cmake/2011-October/046706.html)

the correct way is
```
if( NOT DEFINED ENV{HIP_PATH})
    set( HIP_PATH "/opt/rocm/hip" )
else( )
    set (HIP_PATH $ENV{HIP_PATH} )
endif( )
```
(note the missing dollar in the if)

the consequence of this bug is that building hipSOLVER is not as straightforward as executing `cmake ..` and `make` when HIP is installed in a non-default location
The workaround seems to be adding an additional flag `-DCMAKE_MODULE_PATH="${HIP_PATH}/cmake"` to cmake

Setting `hip_DIR` nor `HIP_DIR` did not help in my case.

Deleting the whole if/else did not help either. The error

```
CMake Error at CMakeLists.txt:98 (find_package):
  No "FindHIP.cmake" found in CMAKE_MODULE_PATH.
```

still persists.

CMake-ing hipSPARSE (current branch develop) using just `cmake -DBUILD_CUDA=ON ..` fails, coult not find amd_comgr (I think amd_comgr should not be needed on nvidia platform). The whole error message:
```
CMake Error at /home/jakub/apps/cmake-3.20.5/share/cmake-3.20/Modules/CMakeFindDependencyMacro.cmake:47 (find_package):
  By not providing "Findamd_comgr.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "amd_comgr", but CMake did not find one.

  Could not find a package configuration file provided by "amd_comgr" with
  any of the following names:

    amd_comgrConfig.cmake
    amd_comgr-config.cmake

  Add the installation prefix of "amd_comgr" to CMAKE_PREFIX_PATH or set
  "amd_comgr_DIR" to a directory containing one of the above files.  If
  "amd_comgr" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  /home/jakub/apps/HIP/installation/lib/cmake/hip/hip-config.cmake:155 (find_dependency)
  cmake/Dependencies.cmake:30 (find_package)
  CMakeLists.txt:98 (include)
```
Setting the `-DCMAKE_MODULE_PATH="${HIP_PATH}/cmake"` solves that, as CMake then reads ${HIP_PATH}/cmake/FindHIP.cmake which does not reference amd_comgr.
It seems similar, if not the same, as I had with rocRAND [https://github.com/ROCmSoftwarePlatform/rocRAND/issues/196]()
Then, executing `make` fails with several compiler errors. branch rocm-4.2.x has similar error, but `make` is fine.

CMake-ing hipFFT (current develop branch) using just `cmake -DBUILD_WITH_LIB=CUDA ..` yields almost the same error as above. Again, using `-DCMAKE_MODULE_PATH="${HIP_PATH}/cmake"` solves that. `make` and `make install` are then fine.

but these errors seem not to be related to the `HIP_PATH` variable.


PS: I would like to suggest unifying the options USE_CUDA, BUILD_CUDA, BUILD_WITH_LIB=CUDA, etc. across the hip* libraries, and maybe set its default depending on HIP_PLATFORM environment variable.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,6 +9,9 @@ Full documentation for hipSOLVER is available at [hipsolver.readthedocs.io](http
 ### Deprecated
 ### Removed
 ### Fixed
+- Fixed use of incorrect `HIP_PATH` when building from source (#40).
+  Thanks [@jakub329homola](https://github.com/jakub329homola)!
+
 ### Known Issues
 ### Security
 

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -55,7 +55,7 @@ include( ROCMInstallSymlinks )
 set ( VERSION_STRING "1.2.0" )
 rocm_setup_version( VERSION ${VERSION_STRING} )
 
-if( NOT DEFINED $ENV{HIP_PATH})
+if( NOT DEFINED ENV{HIP_PATH})
     set( HIP_PATH "/opt/rocm/hip" )
 else( )
     set (HIP_PATH $ENV{HIP_PATH} )

